### PR TITLE
go 1.8.1

### PR DESCRIPTION
**Pull request by mblair**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/go.rb [hunk 1]
```diff
@@ -1,13 +1,11 @@
 class Go < Formula
   desc "The Go programming language"
   homepage "https://golang.org"
-  revision 1
 
   stable do
-    url "https://storage.googleapis.com/golang/go1.8.src.tar.gz"
-    mirror "https://fossies.org/linux/misc/go1.8.src.tar.gz"
```
Comment (neutric) [inline Formula/go.rb]: Please update the mirror URL as if it is already valid.

**Review by neutric — CHANGES_REQUESTED** [review]

File: Formula/go.rb (hunk 1 above)
Comment (mblair) [inline Formula/go.rb]: Done! I also removed the no-longer-necessary patching.

**Comment by neutric** [comment]
Wow, you are fast. I was just going to point out the patches removed in #12202. :)

**Review by neutric — APPROVED** [review]

**Comment by cblecker** [comment]
you may also need to bump the revision of the godep formula in a separate commit

**Comment by mblair** [comment]
@cblecker like [so](abce495e7d2ac0ef198f59cdfa82f5c10b50d378)?

**Comment by cblecker** [comment]
@mblair yup, that should pass tests when CI kicks in

**Comment by DomT4** [comment]
@ilovezfs I got fed up with `godep` and did some poking. It's hardcoding the Cellar path to `go` about a billion times because `go` itself does the same thing, which is presumably getting passed down the chain. I think if we did:

```diff
diff --git a/Formula/go.rb b/Formula/go.rb
index cacb9e4c15..b39f193c7a 100644
--- a/Formula/go.rb
+++ b/Formula/go.rb
@@ -94,6 +94,14 @@ class Go < Formula
         bin.install_symlink libexec/"bin/godoc"
       end
     end
+
+    %W[
+      #{libexec}/pkg/bootstrap/src/bootstrap/cmd/internal/obj/zbootstrap.go
+      #{libexec}/src/cmd/internal/obj/zbootstrap.go
+      #{libexec}/src/runtime/internal/sys/zversion.go
+    ].each do |file|
+      inreplace file, libexec, opt_libexec
+    end
   end
 
   def caveats; <<-EOS.undent
```
That would solve the having to bump `godep` every single time `go` gets touched. I'm pretty sure `runtime` is the one that gets imported widely, so an even smaller `inreplace` job might work.

**Comment by mblair** [comment]
@DomT4 should I apply that patch to this PR?

**Comment by ilovezfs** [comment]
>@DomT4 should I apply that patch to this PR?

No.

**Comment by DomT4** [comment]
It's a significant change, no point attaching it to something that people are desperate to get merged.

**Comment by ilovezfs** [comment]
@DomT4 yeah I'm aware that's the cause. I think we may want to just point the relevant go variables (probably GOROOT in this case) at the opt_prefix during the `go` build. This may require some `brew` back end change as well since the opt link doesn't exist during the build currently, and attempting to create it is a sandbox violation.

**Comment by DomT4** [comment]
> (probably GOROOT in this case)

Yes, indeed. It's being hardcoded as `defaultGOROOT`.

> This may require some brew back end change as well since the opt link doesn't exist during the build currently, and attempting to create it is a sandbox violation.

Yeah, that's what I was effectively trying to avoid with this suggestion. It's something that can be implemented purely on the formula side of things rather than trying to push through likely controversial design changes to `brew` itself.

**Comment by ilovezfs** [comment]
I'd rather it be fixed properly than with inreplace regardless of controversy.

**Comment by DomT4** [comment]
Alright. I'll graciously leave that one down to you 😉.

**Comment by ilovezfs** [comment]
In any case, CI appears to be toast due to this Jenkinsfile pipeline stuff, so people will just need to be patient here.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Shipped! Thanks for the PR @mblair

**Comment by ilovezfs** [comment]
@DomT4 I think this works:
```
iMac-TMP:homebrew-core joe$ git diff -- Formula/go.rb
diff --git a/Formula/go.rb b/Formula/go.rb
index 3ddc8debf0..ff79f8309b 100644
--- a/Formula/go.rb
+++ b/Formula/go.rb
@@ -47,7 +47,7 @@ class Go < Formula
     ENV["GOROOT_BOOTSTRAP"] = buildpath/"gobootstrap"
 
     cd "src" do
-      ENV["GOROOT_FINAL"] = libexec
+      ENV["GOROOT_FINAL"] = opt_libexec
       ENV["GOOS"]         = "darwin"
       ENV["CGO_ENABLED"]  = "0" if build.without?("cgo")
       system "./make.bash", "--no-clean"
```
with
```
diff --git a/Library/Homebrew/formula_installer.rb b/Library/Homebrew/formula_installer.rb
index 017be51dc..fad755bdf 100644
--- a/Library/Homebrew/formula_installer.rb
+++ b/Library/Homebrew/formula_installer.rb
@@ -660,6 +660,9 @@ class FormulaInstaller
 
     Sandbox.print_sandbox_message if Sandbox.formula?(formula)
 
+    formula.prefix.mkpath
+    Keg.new(formula.prefix).optlink
+
     Utils.safe_fork do
       # Invalidate the current sudo timestamp in case a build script calls sudo
       system "/usr/bin/sudo", "-k"
@@ -687,6 +690,7 @@ class FormulaInstaller
   rescue Exception
     ignore_interrupts do
       # any exceptions must leave us with nothing installed
+      Keg.new(formula.prefix).remove_opt_record
       formula.update_head_version
       formula.prefix.rmtree if formula.prefix.directory?
       formula.rack.rmdir_if_possible
```

**Comment by DomT4** [comment]
@ilovezfs Yes, that looks like it should work. Haven't tested it but the idea looks good.

Theoretically that would possibly also allow configuring things with the prefix set to `opt_prefix` instead of `prefix`, resolving some of the complaints that Homebrew `pkgconfig` files almost always point to a versioned `Cellar` rather than a broader persistent directory.

**Comment by ilovezfs** [comment]
Yes, although I think we'd want to limit that from being applied outside of the particular cases that need it otherwise invoking other versions of a formula through their full Cellar path will end up using things from the opt-linked version, which would be a shame.

**Pull request closed without merging**